merlin/models/tf/loader.py

```diff
@@ -289,293 +276,85 @@ def __init__(
         )
         if schema:
             dataset.schema = schema
+
+        cat_names = cat_names or (
+            dataset.schema.select_by_tag(Tags.CATEGORICAL)
+            .excluding_by_tag(Tags.TARGET)
+            .column_names
+            if _get_schema(dataset)
+            else []
+        )
+        cont_names = cont_names or (
+            dataset.schema.select_by_tag(Tags.CONTINUOUS).excluding_by_tag(Tags.TARGET).column_names
+            if _get_schema(dataset)
+            else []
+        )
+        label_names = label_names or (
+            dataset.schema.select_by_tag(Tags.TARGET).column_names if _get_schema(dataset) else []
+        )
+
         cat_names, cont_names, label_names = _validate_schema(
             feature_columns, cat_names, cont_names, label_names, schema=dataset.schema
         )
 
+        dataset.schema = _augment_schema(
+            dataset.schema,
+            cat_names,
+            cont_names,
+            label_names,
+            sparse_names,
+            sparse_max,
+            sparse_as_dense,
+        )
+
         device = "cpu" if not HAS_GPU else device
         if hvd_installed and hvd.size() > 1:
             device = hvd.local_rank()
             global_size = global_size or hvd.size()
             global_rank = global_rank or hvd.rank()
             seed_fn = seed_fn or get_default_hvd_seed_fn()
-        DataLoader.__init__(
-            self,
+
+        super().__init__(
             dataset,
             batch_size,
-            shuffle,
-            cat_names=cat_names,
-            cont_names=cont_names,
-            label_names=label_names,
+            shuffle=shuffle,
             seed_fn=seed_fn,
             parts_per_chunk=parts_per_chunk,
             device=device,
             global_size=global_size,
             global_rank=global_rank,
             drop_last=drop_last,
-            sparse_names=sparse_names,
-            sparse_max=sparse_max,
-            sparse_as_dense=sparse_as_dense,
         )
-        self._transforms = [("all", transform)] if transform else []
-        self.multi_label_as_dict = multi_label_as_dict
-
-    def __len__(self):
-        """
-        recreating since otherwise Keras yells at you
-        """
-        # TODO: what's a better way to do this inheritance
-        # of the appropriate methods? A Metaclass?
-        DataLoader.stop(self)
-        return DataLoader.__len__(self)
 
     def on_epoch_end(self):
```

Comment:
This method can be removed too, now that the loader doesn't inherid from the keras Sequence class